### PR TITLE
[DoNotMerge] Codesigning extension

### PR DESCRIPTION
**Pull request by amritas**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/codesigning/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Azure CLI Codesigning Extension #
+This is an extension to Azure CLI to manage Codesigning resources.
+
+## How to use ##
```
Comment (AllyW) [inline src/codesigning/README.md:4]: please add code usage for users

File: src/codesigning/azext_codesigning/aaz/latest/code_signing/_create.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "code-signing create",
+    is_experimental=True,
+)
+class Create(AAZCommand):
+    """Create a Code Signing Account
```
Comment (AllyW) [inline src/codesigning/azext_codesigning/aaz/latest/code_signing/_create.py:19]: for create/update cmd, please add code examples using codegen tool: add in corresponding cmd's workspace and then export.

File: src/codesigning/azext_codesigning/aaz/latest/code_signing/_update.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "code-signing update",
+    is_experimental=True,
+)
+class Update(AAZCommand):
+    """Update a Code Signing Account
```
Comment (AllyW) [inline src/codesigning/azext_codesigning/aaz/latest/code_signing/_update.py:19]: for create/update cmd, please add code examples using codegen tool: add in corresponding cmd's workspace and then export.

File: src/codesigning/azext_codesigning/aaz/latest/code_signing/certificate_profile/_create.py [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "code-signing certificate-profile create",
+    is_experimental=True,
+)
+class Create(AAZCommand):
+    """Create a certificate profile
```
Comment (AllyW) [inline src/codesigning/azext_codesigning/aaz/latest/code_signing/certificate_profile/_create.py:19]: for create/update cmd, please add code examples using codegen tool: add in corresponding cmd's workspace and then export.

File: src/codesigning/azext_codesigning/aaz/latest/code_signing/certificate_profile/_update.py [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+class Update(AAZCommand):
+    """Update a certificate profile
```
Comment (AllyW) [inline src/codesigning/azext_codesigning/aaz/latest/code_signing/certificate_profile/_update.py:15]: for create/update cmd, please add code examples using codegen tool: add in corresponding cmd's workspace and then export.

File: src/codesigning/azext_codesigning/tests/latest/test_codesigning.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.testsdk import *
+
+
+class CodesigningScenario(ScenarioTest):
+    # TODO: add tests here
```
Comment (AllyW) [inline src/codesigning/azext_codesigning/tests/latest/test_codesigning.py:12]: tests

**Pull request closed without merging**